src/shared/components/FilterList/InjectionOption.tsx

```diff
@@ -30,25 +31,39 @@ interface Props {
   setHoverdFunction?: (string: string) => void
 }
 
-const defaultProps = {
-  testID: 'flux-injection-option',
-}
-
 const FluxInjectionOption: FC<Props> = ({
   option,
   extractor,
   onClick,
-  testID,
+  testID = 'flux-injection-option',
   ToolTipContent,
   searchTerm,
   setToolTipPopup,
   setHoverdFunction,
 }) => {
+  // TODO: move this to a flag once design wants to branch it out further
+  const {fluxQueryBuilder} = useContext(AppSettingContext)
   const itemRef = createRef<HTMLDListElement>()
   const handleClick = () => {
     onClick(option)
   }
 
+  const fullClick = () => {
+    if (!fluxQueryBuilder) {
+      return
+    }
+    onClick(option)
+  }
+
+  const classer = [
+    ['flux-toolbar--list-item', true],
+    ['flux-toolbar--function', true],
+    ['flux-toolbar--new-style', fluxQueryBuilder],
+  ]
+    .filter(c => c[1])
+    .map(c => c[0])
+    .join(' ')
```

Comment:
`classer` is a nice way to reduce a lot of code, but we need signon (for this new pattern) from dumplings. 

Since this is shared code, and we are effectively collapsing several shared components into single DRY components --> but with different styling.

Reply:
this pattern has been in the codebase for over a year, the one that comes to mind is from this PR:
https://github.com/influxdata/ui/pull/2662/files#diff-0823902a1cc6d2e5d230a5665bb76c1ab24842cabb58ebe322346eaa4290a167R106

but in it's essence it's the same pattern as 'classnames' without the library import